pycue/opencue/cloud/gce_api.py

```diff
@@ -0,0 +1,145 @@
+#  Copyright (c) 2018 Sony Pictures Imageworks Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+from .api import CloudInstanceGroup
+
+from googleapiclient import discovery
+from oauth2client.client import GoogleCredentials
+
+
+# TODO : to be replaced with a better way to handle authentication
+credentials = GoogleCredentials.get_application_default()
+service = discovery.build('compute', 'v1', credentials=credentials)
+project = 'gsoc-opencue-test-bed'
+zone = 'us-central1-a'
```

Comment:
Let's avoid having these values as globals. The main problem here is that by importing this file/class, you automatically trigger some code logic to run (fetching credentials and building the service) which can lead to some unexpected behavior and make the class difficult to test.

Normally the pattern would be to have them as arguments to the constructor. Then in the "real" code, pass in the values you have here. Then in the test code, you could pass in whatever values you wanted -- mocks/dummy values/etc.

That won't really work here, because we're implementing an abstract class and we want the constructor params to be the same for different cloud providers. For example the AWS version of this class probably won't use the same project/zone setup, so we don't want to use those parameters.

How about something like this -- move these values to be class-level variables in the `GoogleCloudGroup` class. Add a `connect` method to the class which defines their values. That way the GUI code can instantiate the class and call `connect()`, and the test code can skip calling `connect()`, mocking the values instead.

Reply:
I revisited the way I have setup the CloudGroup class. Since connect() is more of a CloudManager job I have re-structured the methods a bit and pulled out these globals into attributes of the implemented provider Manager class. 
I will discuss this restructuring a bit tomorrow during the check-in but like you mentioned it helped me avoid any google related code completely while setting up tests